internal/pkg/source/source_test.go

```diff
@@ -53,127 +60,56 @@ func (c *testConfig) IsSourceFieldAddr(field *ssa.FieldAddr) bool {
 	return match
 }
 
-type analyzerResult struct {
-	allocations []*ssa.Alloc
-	calls       []*ssa.Call
-	fieldAddr   []*ssa.FieldAddr
-	store       []*ssa.Store
-}
-
 var testAnalyzer = &analysis.Analyzer{
-	Name:       "source",
-	Run:        runTest,
-	Doc:        "test harness for the logic related to sources",
-	Requires:   []*analysis.Analyzer{buildssa.Analyzer},
-	ResultType: reflect.TypeOf(analyzerResult{}),
+	Name:     "source",
+	Run:      runTest,
+	Doc:      "test harness for the logic related to sources",
+	Requires: []*analysis.Analyzer{buildssa.Analyzer},
 }
 
 func runTest(pass *analysis.Pass) (interface{}, error) {
 	in := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA)
-	var result analyzerResult
-	for _, fn := range in.SrcFuncs {
-		for _, b := range fn.Blocks {
-			for _, i := range b.Instrs {
-				switch v := i.(type) {
-				case *ssa.Alloc:
-					result.allocations = append(result.allocations, v)
-				case *ssa.Call:
-					result.calls = append(result.calls, v)
-				case *ssa.FieldAddr:
-					result.fieldAddr = append(result.fieldAddr, v)
-				case *ssa.Store:
-					result.store = append(result.store, v)
-				}
+	config := &testConfig{
+		sourcePattern:      ".foo",
```

Comment:
Nit: Since it's a regex pattern, you mean `\.foo`?  Or possibly `\bfoo`.

Reply:
Good question. 
I am not certain, but it seems that the test framework considers the pattern as a regex when it is surrounded by ``.

Reply:
[That's correct](https://pkg.go.dev/golang.org/x/tools/go/analysis/analysistest?tab=doc#Run) but not what I meant here.

With `sourcePattern` being fed into `regexp.MatchString(c.sourcePattern, d.String())`, I read that as you wanting to match `<any package>.foo`.  Which is to say, I read that dot as intended to match the separator between package and Named, rather than a regexp match-any dot.

Reply:
Ah, good point.
Fixed.